test/model.test.js

```diff
@@ -8860,6 +8860,23 @@ describe('Model', function() {
       assert.equal(error.errors['docArr.0.num'].name, 'CastError');
     });
   });
+
+  it('works if passing class that extends Document to `loadClass()` (gh-12254)', async function() {
+    const DownloadJobSchema = new mongoose.Schema({ test: String });
+
+    class B extends mongoose.Document {}
+
+    B.prototype.foo = 'bar';
+
+    DownloadJobSchema.loadClass(B);
+
+    const Test = db.model('Test', DownloadJobSchema);
+
+    const { _id } = await Test.create({ test: 'value' });
+    const doc = await Test.findById(_id);
+    assert.ok(doc);
+    assert.equal(doc.test, 'value');
```

Comment:
Perhaps we need to assert that the value from the loaded class reflects on the documents.

```suggestion
    assert.equal(doc.foo, 'bar');
    assert.equal(doc.test, 'value');
```